sycl/doc/extensions/Barrier/Barrier.asciidoc

```diff
@@ -0,0 +1,290 @@
+= SYCL_EXT_ONEAPI_BARRIER
```

Comment:
I guess we need to rename the document to `sycl_ext_oneapi_barrier.asciidoc` and move from `Barrier` to `supported` directory.
Tagging @gmlueck to confirm. 
I was hoping to find some guidance in https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc, but I was not able to find recommendations for the file naming.

Reply:
There is an explanation of naming in the recently added [README-process.md](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/README-process.md) document.  The name of the file, the name of the extension, and the name of the feature-test macro must all agree.  Assuming you want to keep the existing feature-test macro, the name of the extension should be "sycl_ext_oneapi_cuda_async_barrier" and the name of the file should be "sycl_ext_oneapi_cuda_async_barrier.asciidoc".  The file should reside in either "supported", "proposed", or "experimental" according to its status.

Please also change the specification document to follow the specification template [template.asciidoc](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc).  The format looks close now, but ther boilerplate of the first few sections should be changed.

Reply:
done